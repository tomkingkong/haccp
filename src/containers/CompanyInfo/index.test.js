import React from 'react';
import { shallow } from 'enzyme';

import { CompanyInfo, mapStateToProps, mapDispatchToProps } from '.';

describe('CompanyInfo', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CompanyInfo />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default states', () => {
		const expected = {
			id: '',
      name: '', 
      email: '',
      address: '',
      phone: '',
      team_member_1_name: '',
      team_member_1_title: ''
		};
    expect(wrapper.state()).toEqual(expected);
  });

  it('should update state on mount', () => {
    const mockData = {
      id: 1,
      name: 'MeatFactory', 
      email: 'meatFactory@meat.com',
      address: '34 address st',
      phone: '934103451',
      team_member_1_name: 'Lucah',
      team_member_1_title: 'Processor'
    }
    wrapper = shallow(<CompanyInfo companyInfo={mockData}/>);		
    wrapper.instance().componentDidMount();
    expect(wrapper.state()).toEqual(mockData);
  });

  describe('handleSubmit', () => {

  });

  describe('handleChange', () => {
    it('should change state on input or checkbox changes', () => {
      const event = {
        target: {
          name: 'name',
          value: 'Schmeckles'
        }
      };
      wrapper.instance().handleChange(event);
      expect(wrapper.state().name).toEqual('Schmeckles');
    });
  });


  describe('mapStateToProps', () => {
    it('should be provided with a products array from props', () => {
      const mockState = {
        editProduct: 1,
	      ingredients: [
        ],
        companyInfo: {
          id: 1,
          name: 'MeatFactory', 
          email: 'meatFactory@meat.com',
          address: '34 address st',
          phone: '934103451',
          team_member_1_name: 'Lucah',
          team_member_1_title: 'Processor'
        },
        products: [
          { id: 1, name:'MeatSticks' }
        ],
        receiving: [],
        inventory: [],
        processing: [],
        packaging: [],
      };
      const expected = { 
        companyInfo: {
          id: 1,
          name: 'MeatFactory', 
          email: 'meatFactory@meat.com',
          address: '34 address st',
          phone: '934103451',
          team_member_1_name: 'Lucah',
          team_member_1_title: 'Processor'
        }
      };
      const results = mapStateToProps(mockState);
      expect(results).toEqual(expected);
    });
  });
});