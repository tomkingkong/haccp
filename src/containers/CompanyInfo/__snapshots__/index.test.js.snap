// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyInfo should match snapshot 1`] = `
<div>
  <h2>
    Enter your company details
  </h2>
  <button
    onClick={[Function]}
  >
    Next Page
  </button>
</div>
`;
